zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/QuorumPeer.java

```diff
@@ -1032,6 +1035,17 @@ private void loadDataBase() {
             long epochOfZxid = ZxidUtils.getEpochFromZxid(lastProcessedZxid);
             try {
                 currentEpoch = readLongFromFile(CURRENT_EPOCH_FILENAME);
+                if (epochOfZxid > currentEpoch && updating.exists()) {
+                    LOG.info("{} found. The server was terminated after "
+                                    + "taking a snapshot but before updating current "
```

Comment:
IMHO, skip creating update file will not cause any problem. old update file will be just reused. it is a flag, does not contain any value in it. Its effect is same as newly created update file.

Reply:
correct,  It's not guaranteed that we had finished taking a snapshot.
Any suggestion to implement adding suffix and how it will be useful in enduring that snapshot was taken

Reply:
I think there is one scenario where it is sure that snapshot is taken but acceptedEpoch file is not updated
The scenario is as follows:
accepted epoch is first written to temporary file acceptedEpoch.tmp then this file is renamed to acceptedEpoch.
Failure, either because of exception or power-off, in renaming the acceptedEpoch.tmp file will cause server startup error with message "The current epoch, x, is older than the last zxid y"

I think we should handle this scenario. I will create PR.

Reply:
Created PR https://github.com/apache/zookeeper/pull/1664